rust/relay/src/server.rs

```diff
@@ -289,6 +296,59 @@ where
         Ok(())
     }
 
+    /// Handle a TURN refresh request.
+    ///
+    /// See <https://www.rfc-editor.org/rfc/rfc8656#name-receiving-a-refresh-request> for details.
+    fn handle_refresh_request(
+        &mut self,
+        message: Message<Attribute>,
+        sender: SocketAddr,
+        now: Instant,
+    ) -> Result<(), ErrorCode> {
+        // TODO: Verify that this is the correct error code.
+        let allocation = self
+            .allocations
+            .get_mut(&sender)
+            .ok_or(ErrorCode::from(AllocationMismatch))?;
+
+        let requested_lifetime = message.get_attribute::<Lifetime>();
+        let effective_lifetime = compute_effective_lifetime(requested_lifetime);
+
+        if effective_lifetime.lifetime().is_zero() {
+            self.pending_commands
+                .push_back(Command::FreeAddresses { id: allocation.id });
+            self.allocations.remove(&sender);
+            self.send_message(
+                refresh_success_response(effective_lifetime, message.transaction_id()),
+                sender,
+            );
+
+            return Ok(());
+        }
+
+        allocation.expires_at = now + effective_lifetime.lifetime();
```

Comment:
Hm, do you think now would be a good time to add a `max_lifetime`? Potentially exposed via an env var for configuration? I think the spec mentions 10 minutes but we discussed extending it last week IIRC.

Reply:
Ah, nvmd -- I see your `DEFAULT_ALLOCATION_LIFETIME` const above. Perhaps a comment here mentioning supporting longer lifetimes would be appropriate for the next reader?

Reply:
> Ah, nvmd -- I see your `DEFAULT_ALLOCATION_LIFETIME` const above. Perhaps a comment here mentioning supporting longer lifetimes would be appropriate for the next reader?

The client can request up to an hour. The default one only applies if no lifetime is explicitly requested.
I think it makes sense to stick to the spec where we can and only deviate once it causes problems.

That that there are two expiring mechanism:

- allocations
- channels

I haven't implemented channels yet. Those are always limited to 10min, I am assuming because they are not authenticated so you want to keep the window relatively short.

I think what we discussed last week was to extend the default lifetime of channels.

Reply:
No worries at all :)